### ISSUE TITLE
Laravel 4.2?

### ISSUE THREAD

**Issue by rombat** [Classy-Geeks/laravel-potion#2]
Is there a way to use it with Laravel 4.2?

**Comment by stevebauman** [comment]
I agree with this. Looking through your code I see no dependencies on Laravel 5 besides some service provider methods. Backwards compatibility can be added in very easily.

Please don't forget about the massive Laravel 4 user base!

**Comment by mattrmiller** [comment]
I will look into the differences between Laravel 4 and Laravel 5. There will probably need to be some changes around how packages, commands, and configs are registered to backwards port to Laravel 4. However, it should be doable. 

Backwards compatibility is always nice. Honestly I did not mean to leave out Laravel 4, this was just written around a Laravel 5 project; and just was never tested/ported to Laravel 4.

**Comment by stevebauman** [comment]
Commands are registered the same, and the only large change from Laravel 4 is that packages now use a typical config separator instead of the package specific one in L4. For example:

```
// L4
Config::get('package::config');


// L5
Config::get('package.config');
```

You can take a look at a service provider I've made for both L4/L5 compatibility here:

https://github.com/stevebauman/inventory/blob/master/src/Stevebauman/Inventory/InventoryServiceProvider.php

All you really need to do is assign a configuration separator and retrieve all your configuration values using it.

If you need help porting I'd gladly do so. I've been looking for this sort of functionality for a while now, but many of my projects are using L4.2. Like you say in your readme, installing Node.js, Gulp, and dependent NPM packages is such a hassle to manage assets.

**Comment by mattrmiller** [comment]
Good stuff. I should be able to get that implemented this week. I might need to create a branch for someone to test with a L4 instance. I don't have one at the moment, but might be able to quickly set one up depending on time.

Thanks!

**Comment by mattrmiller** [comment]
1.0 Fixes this and cleans up some up some other stuff. Thanks for your input and contribution!

**Comment by stevebauman** [comment]
Awesome!

**Comment by rombat** [comment]
Hi,
I can't install the package since it requires illuminate/support: 5.0.\* and it comes with Laravel 5. Can you fix that please?

**Comment by mattrmiller** [comment]
This is fixed in v1.1 I just pushed.

This issue is closed as it pertains to an older version of Potion. If there is any more problems around Laravel compatibility, please open a new issue. 

Thanks.
